runtime/bcutil/ROMClassWriter.cpp

```diff
@@ -180,7 +180,7 @@ class ROMClassWriter::AnnotationWriter :
 
 	void visitAnnotation(U_16 typeIndex, U_16 elementValuePairCount)
 	{
-		_cursor->writeBigEndianU16(_constantPoolMap->getROMClassCPIndexForReference(typeIndex), Cursor::GENERIC);
+		_cursor->writeBigEndianU16(_constantPoolMap->getROMClassCPIndexForAnnotationWithCheck(typeIndex), Cursor::GENERIC);
```

Comment:
Doesn't this still return 0 as the cpIndex for annotations with invalid type_index? I thought that was why the previous PR was closed:
`We realized that 0 is a valid constant pool entry. Closing this while we develop another solution.`

Reply:
The prior solution set the classfile constant pool index to zero (a valid entry).

This change set leaves the classfile constant pool index as-is but does not mark it as used by an annotation.  When we create the ROM constant pool, we check the validity of the  classfile index and force the ROM constant pool index to zero (i.e. not a valid index).   @fengxue-IS please would you please verify my logic?

Reply:
The index zero for classfile CP is the first entry read, where as the zero index in ROM CP is a reserved null entry (which should trigger a runtime exception when attempting to read from it), in the previous PR, we set the classfile cp to zero, so this line of code with attempt to get an value the may be initialized (ie. classfile CP zero entry is not referenced)

Reply:
The zero index in constantpool means null, even in a classfile.  See JVMS 4.1
> The `constant_pool` table is indexed from 1 to `constant_pool_count-1`

Reply:
There is no defined mapping from classfile offset 0, as it is never used, so we felt it was safer to leave a valid classfile index and detect the error when we do the mapping.

Reply:
index 0 is used as a null value.  We're safe to use it here.

Reply:
In that case, I reverted to the previous changeset.